clang/test/Driver/cl-zc.cpp

```diff
@@ -123,6 +123,10 @@
 // CHECK-CHAR8_T_: "-fno-char8_t"
 
 
+// RUN: %clang_cl /dev/null /E -Xclang -dM 2> /dev/null | FileCheck -match-full-lines %s --check-prefix=STDCOFF
+// RUN: %clang_cl /dev/null /E -Xclang -dM /Zc:__STDC__ 2> /dev/null | FileCheck -match-full-lines %s --check-prefix=STDCON
```

Comment:
I'd also like to see a test that ensures you can't *disable* the option. e.g., `-fno-ms-define-stdc` should be an error.

And this current test actually shows a bug -- this is a .cpp file, not a .c file; `/Zc:__STDC__` should do nothing/diagnose in C++ mode (https://godbolt.org/z/6W4oaKf9e); I'm on the fence as to whether `-fms-define-stdc` should be allowed in C++ mode or not (I'm inclined to say it should also be unsupported in C++ mode).

Reply:
Good catch, in this case I guess it's best to create a `cl-zc.c` file?

I think it shouldn't be allowed in clang in C++ mode - after all if someone really wants it they can use `-Xclang -fms-define-stdc` or even `-D__STDC__` manually. My guess would be the primary use case of `-fms-compatibility` with clang (rather than clang-cl) is code whose build system is compiler agnostic but the code itself depends on MSVC behavior - in which case they would not be expecting `__STDC__` to be defined.